onnxruntime/core/providers/cpu/nn/conv_integer.cc

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cpu/nn/conv_integer.h"
+#include "core/util/math.h"
+#include "core/util/math_cpuonly.h"
+#include "public/gemmlowp.h"
+
+namespace onnxruntime {
+
+Status ConvInteger::Compute(OpKernelContext* context) const {
+  size_t num_inputs = OpKernel::Node().InputDefs().size();
+  const Tensor* X = context->Input<Tensor>(0);
+  const Tensor* W = context->Input<Tensor>(1);
+  const Tensor* X_Zero_Point = nullptr;
+  const Tensor* W_Zero_Point = nullptr;
+  int32_t input_offset = 0, filter_offset = 0;
+  if (num_inputs >= 3) {
+    X_Zero_Point = context->Input<Tensor>(2);
+    if (X_Zero_Point->Shape().NumDimensions() == 0 ||
+        (X_Zero_Point->Shape().NumDimensions() == 1 && X_Zero_Point->Shape().GetDims().size() == 1)) {
```

Comment:
Accr to zero_point input def: when zero_point is a 1-D tensor its size should be equal to number of channels in the input... so should we even check this case here? May be just check is zero_point is a scalar... if not then not supported right now